### ISSUE TITLE
Can't parse a JSON encoded message with unknown enum value

### ISSUE THREAD

**Issue by oridag** [protocolbuffers/protobuf#15021]
**What version of protobuf and what language are you using?**
4.22.0
Language: Python

**What runtime / compiler are you using (e.g., python version or gcc version)**
python 3.11

**What did you do?**
Parsed a JSON encoded message which has an unknown enum value

**What did you expect to see**
Message parsed

**What did you see instead?**
```
ValueError: invalid literal for int() with base 10: 'TWO'
```

**Reproducer:**
```
from google.protobuf import descriptor_pb2, timestamp_pb2, empty_pb2, message_factory 
from google.protobuf import json_format       
        
desc_proto = descriptor_pb2.FileDescriptorProto(
    name="f.proto", package="some.package",
    enum_type=[
      descriptor_pb2.EnumDescriptorProto(name='Enum', value=[
        descriptor_pb2.EnumValueDescriptorProto(name='ZERO', number=0),
        descriptor_pb2.EnumValueDescriptorProto(name='ONE', number=1),
        descriptor_pb2.EnumValueDescriptorProto(name='TWO', number=2)
      ])
    ],
    message_type=[
      descriptor_pb2.DescriptorProto(name="ProtoMessage", field=[
        descriptor_pb2.FieldDescriptorProto(name="a_text_field", number=1, type=descriptor_pb2.FieldDescriptorProto.TYPE_STRING),
        descriptor_pb2.FieldDescriptorProto(name="enum", number=2, type=descriptor_pb2.FieldDescriptorProto.TYPE_ENUM, type_name='Enum'),
            
      ])
    ]
)
message_classes = message_factory.GetMessages([desc_proto])
proto_message = message_classes["some.package.ProtoMessage"]
msg = proto_message(a_text_field="hello", enum=2)

json = json_format.MessageToJson(msg)
print(json)

desc_proto.enum_type[0].value.pop()
message_classes = message_factory.GetMessages([desc_proto])
proto_message = message_classes["some.package.ProtoMessage"]


parsed = json_format.Parse(json, proto_message(), ignore_unknown_fields=True)
```

Output
```
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/google/protobuf/json_format.py", line 812, in _ConvertScalarFieldValue
    number = int(value)
             ^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'TWO'
```

**Comment by antongrbin** [comment]
Possibly related: https://github.com/protocolbuffers/protobuf/issues/7392

**Comment by antongrbin** [comment]
I believe this should be fixed in https://github.com/protocolbuffers/protobuf/commit/86abf35ef5ee5b1004ec11bebb36d84c2ef6645e which was released recently as part of 5.27.0.
